src/debuginfo.cpp

```diff
@@ -34,12 +34,12 @@
 #endif
 #ifdef _OS_WINDOWS_
 #include <llvm/Object/COFF.h>
-#   ifdef LLVM37
+#   if JL_LLVM_VERSION >= 30700
 #       include <llvm/Object/ELFObjectFile.h>
 #   endif
 #endif
 
-#if defined(USE_MCJIT) && !defined(LLVM36) && defined(_OS_DARWIN_)
+#if defined(USE_MCJIT) && JL_LLVM_VERSION < 30600 && defined(_OS_DARWIN_)
 #include "../deps/llvm-3.5.0/lib/ExecutionEngine/MCJIT/MCJIT.h"
```

Comment:
The new check below is already doing that so the error message should be updated too